src/main/java/com/google/edith/services/ReceiptFileHandlerService.java

```diff
@@ -0,0 +1,63 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.edith.services;
+
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.FileInfo;
+import com.google.edith.interfaces.ReceiptFileHandlerInterface;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Service that stores file uploaded in Blobstore and serves that blob on succession. */
+public class ReceiptFileHandlerService implements ReceiptFileHandlerInterface {
+  private final BlobstoreService blobstoreService;
+
+  public ReceiptFileHandlerService(BlobstoreService blobstoreService) {
+    this.blobstoreService = blobstoreService;
+  }
+
+  /** Redirects the page to a new tab and serves the blob. */
+  public void serveBlob(HttpServletResponse response, List<FileInfo> fileKeys) throws IOException {
```

Comment:
We can add an @Override annotation for overridden method see:

https://docs.oracle.com/javase/tutorial/java/annotations/predefined.html#:~:text=%40Override%20%40Override%20annotation%20informs%20the,discussed%20in%20Interfaces%20and%20Inheritance.&text=While%20it%20is%20not%20required,it%20helps%20to%20prevent%20errors.

Reply:
Thanks for the reference.
Changes made.